politeiad/backendv2/tstorebe/plugins/pi/hooks.go

```diff
@@ -390,11 +392,256 @@ func (p *piPlugin) voteSummary(token []byte) (*ticketvote.SummaryReply, error) {
+		// the proposal's author.
+		recordAuthorID, err := p.recordAuthor(token)
+		if err != nil {
+			return err
+		}
+
+		if n.UserID == recordAuthorID &&
+			n.ExtraData != "" && n.ExtraDataHint != "" {
+			switch n.ExtraDataHint {
+			case pi.ProposalUpdateHint:
+				// Decode comment extra data
+				var pum pi.ProposalUpdateMetadata
+				err = json.Unmarshal([]byte(n.ExtraData), &pum)
+				if err != nil {
+					return err
+				}
+				// Verify update title
+				if !p.titleIsValid(pum.Title) {
+					return backend.PluginError{
+						PluginID:     pi.PluginID,
+						ErrorCode:    uint32(pi.ErrorCodeUpdateTitleInvalid),
+						ErrorContext: p.titleRegexp.String(),
+					}
+				}
+				// New valid author update
+				return nil
+
+			default:
+				return backend.PluginError{
+					PluginID:  pi.PluginID,
+					ErrorCode: uint32(pi.ErrorCodeExtraDataHintInvalid),
+				}
+
+			}
+		}
+		// New comment is a new thread but not a valid author update.
+		return backend.PluginError{
+			PluginID:     pi.PluginID,
+			ErrorCode:    uint32(pi.ErrorCodeVoteStatusInvalid),
+			ErrorContext: "vote has ended; comments are locked",
+		}
+
+	case isUpdateReply:
+		// This is a reply to the latest update. This is allowed.
+		return nil
+
+	case !isUpdateReply:
+		// New comment is a reply, but is not a reply to the latest update. This
+		// is not allowed.
+		return backend.PluginError{
+			PluginID:  pi.PluginID,
+			ErrorCode: uint32(pi.ErrorCodeWritesAllowedOnlyOnUpdates),
+		}
+
+	default:
+		// This should not happen
+		return errors.Errorf("unknown comment write state")
+	}
+}
+
+// writesAllowedOnApprovedProposal verifies that the given comment write
+// is allowed on the proposal which finished voting, it's vote was approved
+// and is associated with the provided token.
+// This includes both comments and comment votes.
+func (p *piPlugin) writesAllowedOnApprovedProposal(token []byte, cmd, payload string) error {
+	// Get billing status to determine whether to allow author updates
+	// or not.
+	var bsc pi.BillingStatusChange
+	billingStatuses, err := p.billingStatuses(token)
+	if err != nil {
+		return err
+	}
+	// We assume here that admins can set a billing status only once
+	if len(billingStatuses) > 0 {
+		bsc = billingStatuses[0]
+		if bsc.Status == pi.BillingStatusClosed ||
+			bsc.Status == pi.BillingStatusCompleted {
+			// If billing status is set to closed or completed, comment writes
+			// are not allowed.
+			return backend.PluginError{
+				PluginID:  pi.PluginID,
+				ErrorCode: uint32(pi.ErrorCodeBillingStatusInvalid),
+				ErrorContext: "billing status is set to closed/completed;" +
+					" proposal is locked",
+			}
+		}
+	}
+
+	// Get latest proposal author update
+	gar, err := p.comments(token)
+	if err != nil {
+		return err
+	}
+	latestAuthorUpdate, err := latestAuthorUpdate(token, gar.Comments)
+	if err != nil {
+		return err
+	}
+
+	switch cmd {
+	// If that's a new comment then it must be either a new author
```

Comment:
I don't know what `that's` is refering to here.  It's confusing.  Be specific.

Ex.  If the user is submitting a new comment....